spec/models/support_interface/providers_filter_spec.rb

```diff
@@ -1,31 +1,96 @@
 require 'rails_helper'
 
 RSpec.describe SupportInterface::ProvidersFilter do
-  let!(:provider_with_synced_courses) { create(:provider, sync_courses: true) }
-  let!(:provider_without_synced_courses) { create(:provider, sync_courses: false) }
-  let!(:provider_with_signed_dsa) { create(:provider, :with_signed_agreement) }
-  let!(:provider_with_nothing) { create(:provider) }
-
   describe '#filter_records' do
     it 'filters by synced courses' do
+      provider_with_synced_courses = create(:provider, sync_courses: true)
+      create(:provider, sync_courses: false)
+
       providers = Provider.all
       filter = described_class.new(params: { onboarding_stages: %w[synced] })
 
       expect(filter.filter_records(providers)).to eq [provider_with_synced_courses]
     end
 
     it 'filters by DSA signed' do
+      provider_with_signed_dsa = create(:provider, :with_signed_agreement)
+      create(:provider)
+
       providers = Provider.all
       filter = described_class.new(params: { onboarding_stages: %w[dsa_signed] })
 
       expect(filter.filter_records(providers)).to eq [provider_with_signed_dsa]
     end
 
-    it 'excludes providers when the boxes aren’t ticked' do
+    it 'can ignore DSA and synced status' do
```

Comment:
It's a bit unexpected that ticking no boxes in the onboarding status is actually filtering out results. I don't think any of our other filters do that. I guess the need is for a user to be able to filter to see who has done no setup. Would it be weird to change the wording to be "providers who haven't done the thing" since that is what is interesting here?

Reply:
100% agree! I find these checkboxes a bit painful, but they are useful. And `Any` is definitely a hack.

Unchecked checkboxes would still be filtering out results, even if we reversed their meaning.

I thought radio buttons might be better — `Any`, `Courses synced and DSA signed`, and `Courses synced but DSA not signed`?

Soon we'll sync all courses so this will become a bit simpler.

Reply:
If we reversed the meaning of the checkboxes, couldn't it work like: Nothing checked – show all, Check "Haven't signed the DSA" – Filter down to providers who haven't signed it, Check "Haven't synced courses" – Filter down to providers who do not have their courses synced. I think the any is implicit when nothing is checked (at least in our implementation of filters)

Not too important, but that feels more consistent with other filter use to me.

Reply:
We used to have a "synced" and "unsynced" providers tab. That was a bit annoying as you had to flip between them to find providers. It was good, thought, not to have to look at the unsynced ones all the time though as there were lots of them, they didn't tell us much, and they made the list very long.

re DSAs, the original need from the Support board was

```
we need to know when a provider has signed the DSA so that we can open their courses. At the moment, we have to manually check which providers have signed the DSA by going into each individual provider. This is time consuming. 
```

So I wanted to enable filtering by having-DSA too, which is what led me to the current implementation.

How about we do as you say — ie filters behave more like elsewhere — but we keep them positive but use the word "only", ie:

- synced courses only
- signed DSA only

Reply:
Oh I see, that solution makes sense then!